### PR TITLE
DEVOPS-564 docker setup and sync

### PR DESCRIPTION
**Pull request by andytson-inviqa**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,7 +1,9 @@
 source "https://rubygems.org"
 
 group :docker do
-  gem 'rake-tasks-docker', :git => 'https://github.com/inviqa/inviqa-rake-tasks-docker.git'
+  gem 'rake-tasks-docker', '~> 0.1.0'
+  gem 'rake-tasks-docker-setup', '~> 0.1.0'
+  gem 'rake-tasks-docker-sync', git: 'git@github.com:inviqa/inviqa-rake-tasks-docker-sync.git'
```
Comment (andytson-inviqa) [inline Gemfile]: I can't push this one to rubygems repo due to missing teams on the github repo. HELPDESK-6561 raised

File: README.project.md.erb [hunk 2]
```diff
@@ -12,50 +12,74 @@ git clone <%= config.git_url %>
...
+rake docker:setup
+docker-compose up -d
+```
+
+or you can use the rake-tasks-docker helper to setup and up in one go via:
 
+```bash
+bundle install
+
+rake docker:up
```
Comment (rgpjones) [inline README.project.md.erb]: This doesn't seem to run `docker-compose up` in a detached state.

Comment (rgpjones) [reply README.project.md.erb]: It also ended in a "Project failed to start" message, despite the web container still building.

File: README.project.md.erb [hunk 3]
```diff
@@ -12,50 +12,74 @@ git clone <%= config.git_url %>
...
     - On Linux/OSX: `echo "127.0.0.1 <%= config.hostname %>" | sudo tee -a /etc/hosts`
     - On Windows, open up `C:\Windows\system32\etc\HOSTS` in your favourite text editor as admin and add `127.0.0.1 <%= config.hostname %>`
-10. Run `dockrsync watch` to watch for changes to your local filesystem and upload them to the web container.
-11. The site will be available at [https://<%= config.hostname %>](https://<%= config.hostname %>). Happy coding!
+5. Setup docker-compose.override config and start docker-compose up:
+
+```bash
+bundle install
+
+rake docker:setup
```
Comment (kierenevans) [inline README.project.md.erb]: This gives:
```
$ rake docker:setup
rake aborted!
Don't know how to build task 'docker:start' (see --tasks)
/Users/kevans/.hem/gems/ruby/2.2.0/bundler/gems/inviqa-rake-tasks-docker-sync-e1dae4989882/lib/rake-tasks-docker-sync/rake_task.rb:59:in `<top (required)>'
/Users/kevans/Repositories/DEVOPS-564-docker-setup-and-sync/Rakefile:3:in `require'
/Users/kevans/Repositories/DEVOPS-564-docker-setup-and-sync/Rakefile:3:in `<top (required)>'
/Users/kevans/.hem/gems/ruby/2.2.0/gems/rake-12.0.0/exe/rake:27:in `<top (required)>'
/usr/local/bin/ruby_executable_hooks:15:in `eval'
/usr/local/bin/ruby_executable_hooks:15:in `<main>'
(See full trace by running task with --trace)
```

File: README.project.md.erb (hunk 2 above)
Comment (andytson-inviqa) [reply README.project.md.erb]: discussed offline yesterday.

rake docker:up starts in detached and immediately logs. Waits for health checks to pass before giving an indication if the project appears to be working, then closes the log

It's meant as a way to help debug startup issues without forever blocking. If there are no health-checks, it doesn't open the logs and returns immediately.

File: Gemfile (hunk 1 above)
Comment (andytson-inviqa) [reply Gemfile]: this will need updating when we push the updates to the gemss

File: README.project.md.erb (hunk 3 above)
Comment (andytson-inviqa) [reply README.project.md.erb]: The following PR's fix this
https://github.com/inviqa/inviqa-rake-tasks-docker/pull/5
https://github.com/inviqa/inviqa-rake-tasks-docker-sync/pull/1

Comment (kierenevans) [reply README.project.md.erb]: Approved both :+1:, will give another go after they are merged and Gemfile.lock updated

Comment (andytson-inviqa) [reply README.project.md.erb]: done @kierenevans

Comment (kierenevans) [reply README.project.md.erb]: This is now working with delegated mode and sync mode with one hiccup.

As I had `RAKE_USE_DOCKER_SYNC=true` defined in my shell, `docker-compose up -d` from the README didn't work due to:
```
➜  DEVOPS-564-docker-setup-and-sync-2 git:(develop) docker-compose up -d
Creating network "devops564dockersetupandsync2_default" with the default driver
ERROR: Volume DEVOPS-564-docker-setup-and-sync-2-sync declared as external, but could not be found. Please create the volume manually using `docker volume create --name=DEVOPS-564-docker-setup-and-sync-2-sync` and try again.
```

I used the second version, `rake docker:up` and it worked. Do we need to port the RAKE_USE_DOCKER_SYNC documentation from #90 and advocate the `rake docker:up` command by default?

Comment (kierenevans) [reply README.project.md.erb]: RAKE_USE_DOCKER_SYNC docs were on #101 instead of 90.

**Pull request merged**